src/UI/toolbar/BatteryIndicator.qml

```diff
@@ -278,6 +322,117 @@ Item {
                         }
                     }
                 }
+                
+            }
+
+            SettingsGroupLayout {
+                heading:        qsTr("Battery State Display")
+                Layout.fillWidth: true
+                spacing: ScreenTools.defaultFontPixelWidth * 0.5
+                
+                RowLayout {
+                    //Layout.fillWidth: true
+                    
+                    QGCColoredImage {
+                        source: "/qmlimages/BatteryGreen.svg"
+                        height: ScreenTools.defaultFontPixelHeight * 5
+                        width: ScreenTools.defaultFontPixelWidth * 7
+                        fillMode: Image.PreserveAspectFit
+                        color: qgcPal.colorGreen
+                    }
+
+                    QGCLabel {
+                        Layout.preferredWidth: ScreenTools.defaultFontPixelWidth * 3.5 
+                        text: qsTr("100%")
+                        Layout.leftMargin: -13  
```

Comment:
You should never use fix pixels as margins. Everything needs to be based on the current font. Otherwise the ui won't scale to font size changes.

The way to lay this out is as follows:
* Each battery icon plus either fixed percentage or text field should be in its own RowLayout. With the spacing on the layout set to something like font width times 1 or 2 depending on how it looks.
* Then there is a top level RowLayout which has all these RowLayouts as children. And then there is a font based spacing on that to space out the battery+pct visuals.

Reply:
I appreciate your guidance, and I've implemented the layout adjustments as you suggested. Each battery icon, along with its corresponding percentage or text field, is now placed in its own `RowLayout`. I've also set the spacing based on the current font width, ensuring that the UI scales properly with font size changes. The top-level `RowLayout` contains all these individual layouts with appropriate font-based spacing.